pkg/frontend/openshiftcluster_putorpatch_test.go

```diff
@@ -3305,3 +3305,107 @@ func TestEnrichClusterSystemData(t *testing.T) {
 		})
 	}
 }
+
+func TestPersistIdentityUrl(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		identityURL []string
+		cluster     *api.OpenShiftCluster
+		expected    *api.OpenShiftCluster
+		isCreate    bool
+		wantError   string
+	}{
+		{
+			name:        "identity URL is nil, is not wi/mi cluster create",
+			identityURL: nil,
+			cluster:     &api.OpenShiftCluster{},
+			expected:    &api.OpenShiftCluster{},
+			isCreate:    false,
+			wantError:   "",
+		},
+		{
+			name:        "identity URL is nil, is wi/mi cluster create",
+			identityURL: nil,
+			cluster:     &api.OpenShiftCluster{},
+			expected:    &api.OpenShiftCluster{},
+			isCreate:    true,
+			wantError:   "identityURL not provided but required for workload identity cluster",
```

Comment:
Nit: in `openshiftcluster_putorpatch.go`, define an error there so we don't have to repeat the same string twice